### PR TITLE
Feature/unscaled gw cost

### PR DESCRIPTION
**Pull request by michalk8**
TODO:
- [x] fix the expected value in test
- [x] test in FGW

### CODE REVIEW

File: src/ott/problems/quadratic/quadratic_problem.py [hunk 1]
```diff
@@ -285,6 +306,8 @@ def update_linearization(
       epsilon: An epsilon scheduler or a float passed on to the linearization.
       old_transport_mass: Sum of the elements of the transport matrix at the
         previous iteration.
+      remove_scale: Whether to remove any scaling from the cost matrices before
+        computing the linearization.
```
Comment (marcocuturi) [inline src/ott/problems/quadratic/quadratic_problem.py:310]: here I would specify the only case where this makes sense is for the last iteration of the outer loop. e.g.:

remove_scale: Whether to remove any scaling from the cost matrices when
        computing the linearization of the quadratic cost. At the moment, this is
        only used when doing this update at the last outer iteration of a GW solver.

File: src/ott/solvers/quadratic/gromov_wasserstein.py [hunk 2]
```diff
@@ -157,6 +157,9 @@ class GromovWasserstein(was_solver.WassersteinSolver):
     warm_start: Whether to initialize (low-rank) Sinkhorn calls using values
       from the previous iteration. If `None`, warm starts are not used for
       standard Sinkhorn, but used for low-rank Sinkhorn.
+    unscale_last_linearization: Whether to remove any scaling from the
+      cost matrices of the last linearized
+      :attr:`~ott.solvers.quadratic.gromov_wasserstein.GWOutput.geom`
```
Comment (marcocuturi) [inline src/ott/solvers/quadratic/gromov_wasserstein.py:162]: Maybe add for clarity:
unscale_last_linearization: Whether to remove any scaling from the
      cost matrices of the last linearization stored in
      :attr:`~ott.solvers.quadratic.gromov_wasserstein.GWOutput.geom`. This has the practical benefit that, while the OT coupling matrices obtained with GW might have been computed by re-scaling cost matrices for numerical stability, the last linearization stored in :attr:`~ott.solvers.quadratic.gromov_wasserstein.GWOutput.geom` will be un-scaled and recomputed with the original cost values.

**Review by marcocuturi — COMMENTED** [review]
LGTM for now!

File: src/ott/problems/quadratic/quadratic_problem.py (hunk 1 above)
Comment (michalk8) [reply src/ott/problems/quadratic/quadratic_problem.py]: Done!

File: src/ott/problems/quadratic/quadratic_problem.py [hunk 3]
```diff
@@ -155,17 +159,24 @@ def marginal_dependent_cost(
     Args:
       marginal_1: [n,], first marginal of transport matrix.
       marginal_2: [m,], second marginal of transport matrix.
+      remove_scale: Whether to remove any scaling from the cost matrices before
+        computing the linearization.
```
[the remark this answers is not in the document]
Comment (michalk8) [reply src/ott/problems/quadratic/quadratic_problem.py:163]: Done!

File: src/ott/solvers/quadratic/gromov_wasserstein.py (hunk 2 above)
Comment (michalk8) [reply src/ott/solvers/quadratic/gromov_wasserstein.py]: Done!

**Comment by michalk8** [comment]
> LGTM for now!

Thanks, merging this now!

**Pull request merged**